### PR TITLE
Only rewrite database() against dual

### PR DESCRIPTION
**Pull request by systay**

### CODE REVIEW

File: go/vt/sqlparser/expression_rewriting_test.go [hunk 1]
```diff
@@ -55,14 +55,19 @@ func TestRewrites(in *testing.T) {
 			db:       true, liid: true,
 		},
 		{
-			in:       "select (select database() from test) from test",
-			expected: "select (select :__vtdbname as `database()` from test) as `(select database() from test)` from test",
+			in:       "select (select database() from dual) from dual",
+			expected: "select (select :__vtdbname as `database()` from dual) as `(select database() from dual)` from dual",
```
Comment (deepthi) [inline go/vt/sqlparser/expression_rewriting_test.go]: what would happen with a query like:
`select (select database() from dual) from test`
or
`select (select database()) from test`
or even
`select database() from test`

On a side note, the aliases we are producing look weird to me when the SelectExpr is a function or a query.

**Comment by systay** [comment]
Moved to #5793

**Comment by deepthi** [comment]
Reopening to try to fix travis build.

**Comment by deepthi** [comment]
now that #5793 has been merged, we can close this.

**Pull request closed without merging**